emulator_test.go

```diff
@@ -137,6 +140,43 @@ func TestTemperatureEmulationAnomalies_Trend(t *testing.T) {
 	assert.True(t, mean(results) > emulator.T.MeanTemperature)
 }
 
+func TestTemperatureEmulationAnomalies_Trend_Time_Duration(t *testing.T) {
+	emulator := createEmulator(14400, 0)
+	emulator.T.IsTrendAnomaly = true
+	emulator.T.TrendAnomalyMagnitude = 30.0
+	emulator.T.TrendAnomalyLength = 1e3
+	emulator.T.IsRisingTrendAnomaly = true
+
+	emulator.T.TrendAnomalyTimeStart = time.Now().Unix()
+	emulator.T.TrendAnomalyTimeDuration = 20
+
+	var results []float64
+	for continueFlag := true; continueFlag; continueFlag = (emulator.T.TrendAnomalyTimeCurrent - emulator.T.TrendAnomalyTimeStart) < emulator.T.TrendAnomalyTimeDuration {
+		emulator.Step()
+		results = append(results, emulator.T.T)
+		emulator.T.TrendAnomalyTimeCurrent = time.Now().Unix()
+	}
+
+	assert.True(t, mean(results) > emulator.T.MeanTemperature)
+}
```

Comment:
This test is a good example of what I mean by wall time—this takes 20s to run (as we're comparing the `Time` at a given point with a delta defined by the elapsed time). I think what we'd prefer is casting the time to the number of calls to `.Step()`, which means this test need not take 20s to run.